data-structures.md

```diff
@@ -29,22 +29,47 @@ For most objects referenced by Filecoin, a Content Identifier (CID for short) is
 
 CIDs are serialized by applying binary multibase encoding, then encoding that as a CBOR byte array with a tag of 42.
 
+## Tickets
+
+A ticket is Fielcoin's version of a cryptographic lottery ticket and is used to prove that one is a block leader. At a high-level, tickets must do the following:
+
+- Ensure leader secrecy -- meaning a block producer will not be known until they release their block to the network.
+- Prove leader election -- meaning a block producer can be verified by any participant in the network.
+- Prove appropriate delay between drawings — thereby preventing leaders from "rushing" the protocol by releasing blocks early (at the expense of fairness for miners with worse connectivity).
+- Ensure a single drawing per round — derived in part from the above, thereby preventing miners from grinding on tickets (e.g. by repeatedly drawing new tickets in the hopes of winning) within a round.
+
+```go
+type Ticket struct {
+  
+	// The VDF Result is derived from the prior ticket in the ticket chain
+	VDFResult BigInteger
+  
+	// The VDF proves a delay between tickets generated
+	VDFProof BigInteger
+  
+	// This signature is generated by the miner's keypair run on the VDFResult.
+	// It is the value that will be used to generate future tickets or ElectionProofs.
+	Signature SignedMessage
+}
+```
+
 ## Block
 
 A block represents an individual point in time that the network may achieve consensus on. It contains (via merkle links) the full state of the system, references to the previous state, and some notion of a 'weight' for deciding which block is the 'best'.
 
 ```go
 // Block is a block in the blockchain.
 type Block struct {
+  
 	// Miner is the address of the miner actor that mined this block.
 	Miner Address
 
 	// Tickets are the winning ticket that were submitted with this block.
 	Tickets []Ticket
-    
-    // ElectionProof is a signature over the final ticket that proves this miner
-    // is the leader at this round
-    ElectionProof Signature
+
+	// ElectionProof is a signature over the final ticket that proves this miner
+	// is the leader at this round
+	ElectionProof SignatureBytes
```

Comment:
Is it defined somewhere what kind of signature this is cc @sternhenri

Reply:
^ I took a guess at this. Will need @sternhenri to verify